src/lib/optimizer/optimizer.cpp

```diff
@@ -136,9 +77,9 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
   // StoredTableNode as possible where the ChunkPruningRule can work with them.
   optimizer->add_rule(std::make_unique<ChunkPruningRule>());
 
-  // This is an optimization for the PQP sub-plan memoization which is sensitive to the a StoredTableNode's table name,
-  // set of pruned chunks and set of pruned columns. Since this rule depends on pruning information, it has to be
-  // executed after the ColumnPruningRule and ChunkPruningRule.
+  // This is an optimization for better PQP subplan deduplication in the LQPTranslator. It aligns all StoredTableNodes
+  // that refer to the same table and the same set of pruned chunk ids. It is sensitive to all pruning information
+  // and has therefore to be executed after the ColumnPruningRule and the ChunkPruningRule.
```

Comment:
With #2260, the part about chunk pruning becomes incorrect, correct? We could make the alignment rule part of the post-cache optimizer, but that might hurt performance. Otherwise, I wonder if the chunk id distinction will not effectively become dead code with that PR.

Could you discuss this with @j-tr, please?

Reply:
Ok, I'll reach out to him.

Reply:
I had my discussion with @j-tr. We decided to add the rule to the Post-Cache Optimizer. I do not expect a major performance drawback. But we'll see.

Reply:
By reading and thinking about @mrks [comment/suggestion](https://github.com/hyrise/hyrise/pull/2267#discussion_r530184531), I realized that we do not need the ChunkPruningRule to have executed. Great!

So we can put the rule into the Pre-Cache Optimizer :) @j-tr

Reply:
Had a second thought: It is not necessary to execute the ChunkPruningRule before the STNColumnAlignmentRule, BUT it better should be.

For example:
1. STN1 (a)
    STN2 (a, b, c, d)

2. Apply ChunkPruningRule
-> STN1 (a), prunedChunkIDs{0, 1}
-> STN2 (a, b, c, d), prunedChunkIDs{2, 3}

3. Apply STNColumnAlignmentRule
-> STN1 (a), prunedChunkIDs{0, 1}
-> STN2 (a, b, c, d), prunedChunkIDs{2, 3}
=> different sets of pruned chunk ids make pruned column id alignment superfluous

4. LQPTranslator
-> GetTable1 (a), prunedChunkIDs{0, 1}
-> GetTable2 (a, b, c, d), prunedChunkIDs{2, 3}

When switching steps (2) and (3), we get:
-> GetTable1 (a, b, c, d), prunedChunkIDs{0, 1}
-> GetTable2 (a, b, c, d), prunedChunkIDs{2, 3}

=> GetTable1 misses optimization potential. Columns b, c & d could be pruned.

--- 

No idea how the performance will be affected. @j-tr will benchmark both approaches.